src/System.Windows.Extensions/tests/System/Drawing/FontConverterTests.cs

```diff
@@ -105,16 +107,27 @@ public void EmptyStringInput()
                 data.Add($"Courier New{s_Separator} Style=Bold", "Courier New", 8.25f, GraphicsUnit.Point, FontStyle.Bold);
                 data.Add($"11px{s_Separator} Style=Bold", "Microsoft Sans Serif", 8.25f, GraphicsUnit.Point, FontStyle.Bold);
 
-                // FullFramework disregards all arguments if the font name is an empty string.
-                if (PlatformDetection.IsWindows10Version1607OrGreater)
+                // empty string is not an installed font on Windows 7, windows 8 and some versions of windows 10.
+                if (EmptyFontPresent)
+                {
+                    data.Add($"{s_Separator} 10{s_Separator} style=bold", "", 10f, GraphicsUnit.Point, FontStyle.Bold);
+                }
+                else
                 {
-                    data.Add($"{s_Separator} 10{s_Separator} style=bold", "", 10f, GraphicsUnit.Point, FontStyle.Bold); // empty string is not a installed font on Windows 7 and windows 8.
+                    data.Add($"{s_Separator} 10{s_Separator} style=bold", "Microsoft Sans Serif", 10f, GraphicsUnit.Point, FontStyle.Bold);
```

Comment:
Do we need this else branch?

Reply:
The  main purpose of adding this is to show that if the name string is empty then the other attributes like size and style wont be set to default value which is the case with .NetFramework.

Reply:
> is to show that if the name string is empty then the other attributes like size and style wont be set to default value 

Right, that explains the `if` block, but this `else` block isn't testing that.  What is this `else` block testing that's not tested elsewhere?

Reply:
The ```if``` block will get executed only on machines with latest version of windows 10 and ```else``` block will get executed on all machines with windows 7, windows 8 and earlier versions of windows 10

Reply:
But the else block isn't testing the empty string. So how does it differ from all the other tests already being done on non-empty strings?

Reply:
sorry i was not clear. The input string is ```", 10, style=bold"``` . here the name is an empty string, which will automatically converted to ```Microsoft Sans Serif``` by System.Drawing.Font if its not an installed font.

Reply:
Ah!  Ok.  Thanks.